Content.Server/GameObjects/Components/Mobs/ServerAlertsComponent.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using Content.Server.Commands;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Shared.Alert;
+using Content.Shared.GameObjects.Components.Mobs;
+using Robust.Server.Interfaces.Console;
+using Robust.Server.Interfaces.Player;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Interfaces.Network;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Players;
+
+namespace Content.Server.GameObjects.Components.Mobs
+{
+    [RegisterComponent]
+    [ComponentReference(typeof(SharedAlertsComponent))]
+    public sealed class ServerAlertsComponent : SharedAlertsComponent
+    {
+
+        protected override void Startup()
+        {
+            base.Startup();
+
+            if (EntitySystem.TryGet<WeightlessSystem>(out var weightlessSystem))
+            {
+                weightlessSystem.AddAlert(this);
+            }
+            else
+            {
+                Logger.WarningS("alert", "weightlesssystem not found");
+            }
+        }
+
+        public override void OnRemove()
+        {
+            if (EntitySystem.TryGet<WeightlessSystem>(out var weightlessSystem))
+            {
+                weightlessSystem.RemoveAlert(this);
+            }
+            else
+            {
+                Logger.WarningS("alert", "weightlesssystem not found");
+            }
+
+            base.OnRemove();
+        }
```

Comment:
Maybe use eventbus instead?

Reply:
this is not something new by me, this was how it was already implemented in the existing ServerStatusEffectsComponent which I have renamed to ServerAlertsComponent